lib/internal/abort_controller.js

```diff
@@ -243,24 +254,27 @@ class AbortSignal extends EventTarget {
       }
       signal[kDependantSignals] ??= new SafeSet();
       if (!signal[kComposite]) {
-        resultSignal[kSourceSignals].add(new SafeWeakRef(signal));
+        const signalWeakRef = new SafeWeakRef(signal);
+        finalizers.register(resultSignal, signalWeakRef);
```

Comment:
I would place the `finalizers.register` call below the `signal[kDependantSignals].add` call to make it clear they belong together and to have the same order as in the composed case below.